forge-ai/src/main/java/forge/ai/ComputerUtilCombat.java

```diff
@@ -2031,16 +2034,31 @@ public static boolean canDestroyBlocker(Player ai, Card blocker, Card attacker,
      * @param defender
      * @param overrideOrder overriding combatant order
      */
-    public static Map<Card, Integer> distributeAIDamage(final Card attacker, final CardCollectionView block, final CardCollectionView remaining, int dmgCanDeal, GameEntity defender, boolean overrideOrder) {
-        // TODO: Distribute defensive Damage (AI controls how damage is dealt to own cards) for Banding and Defensive Formation
+    public static Map<Card, Integer> distributeAIDamage(final Player self, final Card attacker, final CardCollectionView block, final CardCollectionView remaining, int dmgCanDeal, GameEntity defender, boolean overrideOrder) {
         Map<Card, Integer> damageMap = Maps.newHashMap();
         Combat combat = attacker.getGame().getCombat();
 
         boolean isAttacking = defender != null;
 
+        // Check for Banding, Defensive Formation
+        boolean isAttackingMe = isAttacking && combat.getDefenderPlayerByAttacker(attacker).equals(self);
+        boolean isBlockingMyBand = false;
+        if (attacker.getController().isOpponentOf(self)) {
```

Comment:
isn't this just `AttackingBand.isValidBand`?

Reply:
I think just checking that the attacker(blocker) controller isn't AI should be enough to determine that it should switch from "good" assignments to "bad"?

Reply:
> I think just checking that the attacker(blocker) controller isn't AI should be enough to determine that it should switch from "good" assignments to "bad"?

attacker might be teammate of ai

Reply:
> isn't this just `AttackingBand.isValidBand`?

Pretty much, yeah, will convert to this call.